### PR TITLE
document "addprocs" keyword arg "topology"

### PR DESCRIPTION
**Pull request by amitmurthy**

### CODE REVIEW

File: doc/stdlib/file.rst [hunk 1]
```diff
@@ -337,7 +337,7 @@
 
    .. Docstring generated from Julia source
 
-   ``@__FILE__`` expands to a string with the absolute file path of the file containing the macro. Returns ``nothing`` if run from a REPL or an empty string if evaluated by ``julia -e <expr>``. Alternatively see :data:`PROGRAM_FILE`.
+   ``@__FILE__`` expands to a string with the absolute file path of the file containing the macro. Returns ``nothing`` if run from a REPL or an empty string if evaluated by ``julia -e <expr>``\ . Alternatively see ```PROGRAM_FILE`` <:data:`PROGRAM_FILE`>`_\ .
```
Comment (amitmurthy) [inline doc/stdlib/file.rst]: This change sneaked in upon running `make julia-genstdlib` . I presume it is correct.

File: doc/stdlib/parallel.rst [hunk 2]
```diff
@@ -189,6 +189,21 @@ General Parallel Computing Support
 
    ``exeflags``\ : additional flags passed to the worker processes.
 
+   ``topology``\ : Specifies how the workers connect to each other. Sending a message             between unconnected workers results in an error.
+
+   .. code-block:: julia
```
Comment (tkelman) [inline doc/stdlib/parallel.rst]: I don't think a julia code block is desired here?

File: doc/stdlib/file.rst (hunk 1 above)
Comment (tkelman) [inline doc/stdlib/file.rst]: cc @MichaelHatherly @hayd is this right for cross-references?

File: doc/stdlib/parallel.rst (hunk 2 above)
Comment (amitmurthy) [inline doc/stdlib/parallel.rst]: Well, I would like it indented. Any other way?

File: doc/stdlib/file.rst (hunk 1 above)
Comment (MichaelHatherly) [inline doc/stdlib/file.rst]: At the moment we have to special case the rendering of those, see https://github.com/JuliaLang/julia/blob/4d1bf92752a3efb6370980b722d8832d632ec9fd/base/markdown/render/rst.jl#L86. Just needs `data` added to the regex I think.

File: doc/stdlib/parallel.rst (hunk 2 above)
Comment (tkelman) [inline doc/stdlib/parallel.rst]: I think a markdown bullet list would be appropriate for this?

Comment (amitmurthy) [inline doc/stdlib/parallel.rst]: Sure.

File: base/docs/helpdb/Base.jl [hunk 3]
```diff
@@ -3320,23 +3320,38 @@ it will launch as many workers as the number of cores on the specific host.
...
+  + `:all_to_all`  :  All processes are connected to each other.
+                      This is the default.
+
+  + `:master_slave`  :  Only the driver process, i.e. pid 1 connects to the
+                        workers. The workers do not connect to each other.
+
+  + `:custom`  :  The `launch` method of the cluster manager specifes the
+                  connection topology via fields `ident` and `connect_idents` in
+                  `WorkerConfig`. A worker with a cluster manager identity `ident`
+                  will connect to all workers specified in `connect_idents`.
```
Comment (amitmurthy) [inline base/docs/helpdb/Base.jl]: Would like to have nested lists here. The markdown syntax for the same is not yet supported?

Comment (hayd) [inline base/docs/helpdb/Base.jl]: https://github.com/JuliaArchive/Markdown.jl/issues/8
IIRC @one-more-minute had an idea how to do this cleanly.

(Maybe the Markdown.jl issues should be migrated?)

**Pull request merged**